scripts/sanitycheck.fsx

```diff
@@ -0,0 +1,592 @@
+            =
+            Console.Error.WriteLine(
+                sprintf
+                    "Package found with more than one version: %s. All occurrences:"
+                    key
+            )
+
+            for file, pkgInfo in packageInfos do
+                Console.Error.WriteLine(
+                    sprintf
+                        "* Version: %s. Dependency holder: %s"
+                        pkgInfo.PackageVersion
+                        file.DependencyHolderName.Name
+                )
+
+        let packagesWithMoreThanOneVersion =
+            findPackagesWithMoreThanOneVersion packageTree
+
+        if packagesWithMoreThanOneVersion.Any() then
+            Map.iter pkgWithMoreThan1VersionPrint packagesWithMoreThanOneVersion
+            Environment.Exit 1
+
+        let findPackagesWithSomeReqReinstallAttrib
+            (packageTree: Map<ComparableFileInfo, seq<PackageInfo>>)
+            : seq<ComparableFileInfo * PackageInfo> =
+            seq {
+                for KeyValue(file, packageInfos) in packageTree do
+                    for pkg in packageInfos do
+                        match pkg.ReqReinstall with
+                        | Some true -> yield file, pkg
+                        | _ -> ()
+            }
+
+        let packagesWithWithSomeReqReinstallAttrib =
+            findPackagesWithSomeReqReinstallAttrib packageTree
+
+        if packagesWithWithSomeReqReinstallAttrib.Any() then
+            Console.Error.WriteLine(
+                sprintf
+                    "Packages found with some RequireReinstall attribute (please reinstall it before pushing):"
+            )
+
+            for file, pkg in packagesWithWithSomeReqReinstallAttrib do
+                Console.Error.WriteLine(
+                    sprintf
+                        "* Name: %s. Project: %s"
+                        pkg.PackageId
+                        file.DependencyHolderName.Name
+                )
+
+            Environment.Exit 1
+
+        Console.WriteLine(
+            sprintf
+                "Nuget sanity check succeeded for solution dir %s"
+                solDir.FullName
+        )
+
+
+    let rec findSolutions(dir: DirectoryInfo) : seq<FileInfo> =
+        dir.Refresh()
+
+        seq {
+            // FIXME: avoid returning duplicates? (in case there are 2 .sln files in the same dir...)
+            for file in dir.EnumerateFiles() do
+                if file.Name.ToLower().EndsWith ".sln" then
+                    yield file
+
+            for subdir in dir.EnumerateDirectories().Where notSubmodule do
+                for solution in findSolutions subdir do
+                    yield solution
+        }
+
+    //let solutions = Directory.GetCurrentDirectory() |> DirectoryInfo |> findSolutions
+    //NOTE: we hardcode the solutions rather than the line above, because e.g. Linux OS can't build/restore iOS proj
+    let solutionFileNames =
+        [
+            Path.Combine(ScriptsDir.FullName, "../conventions.sln")
```

Comment:
@Mersho the use of Path.Combine function is **precisely** to combine paths without having to hardcode the slash (which in Windows would be a backslash and in Unix would be a forward slash); if you're not understanding what I'm saying, please ask Afshin (@aarani) to explain to you (either in person or in our telegram channel)